app/Platform/Concerns/BuildsGameListQueries.php

```diff
@@ -151,6 +151,26 @@ private function applyFilters(Builder $query, array $filters, ?User $user = null
                 continue;
             }
 
+            /*
+             * only show games based on their tags
+             */
+            if ($filterKey === 'game-type' && !empty($filterValues)) {
+                if (in_array('retail', $filterValues)) {
```

Comment:
While the UI doesn't allow it, you can pass multiple filter values by modifying the URL. (i.e. http://localhost:64000/system/4-game-boy/games?filter%5Bgame-type%5D=demo%2Chomebrew). This logic assumes that retail excludes all the other values (which may be semantically true, but a user might want to search on retail,demo,prototype and ignore hacks and homebrews).

And on that note, should the UI allow multi-select?

Reply:
Looks like our other single-select filters are affected by this "URL injection" bug as well:

![Screenshot 2025-03-23 at 1 50 40 PM](https://github.com/user-attachments/assets/f13f63a1-8452-4a58-8bc7-1ce8a62dca64)

I lean towards keeping this a single-select right now for two reasons:
* I'm not sure how to handle when users will want a logical AND vs a logical OR, especially if @televandalist's tag revamps go through at some point in the foreseeable future.
* A single-select likely covers 80%+ of the current use cases for the control. If/when people clamor for additional functionality in the filter, we can build it based on feedback.

Reply:
https://github.com/RetroAchievements/RAWeb/pull/3344 resolves the injection bug on single select filters.